### PR TITLE
fix(@ngtools/webpack): skip non-runtime types when transforming constructors

### LINKED ISSUE #14876: Error when using exported interface
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [ ] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

 Yes, the previous version in which this bug was not present was: 8.0.3
(See "anything else relevant" section, exact bug has appeared and been fixed before)

### Description

 A clear and concise description of the problem...
Using an exported interface produces an error, app is not served.

## 🔬 Minimal Reproduction

export an interface:
```
export interface AppConfig {..}
export let APP_CONFIG = new InjectionToken<AppConfig>('app.config');
```
Import it elsewhere:
`import { APP_CONFIG, AppConfig } from '../app.config';`
Use it:
```
export class MyService {
  constructor(@Inject(APP_CONFIG) config: AppConfig,...
  ...
}
```
## 🔥 Exception or Error
ng serve
<pre><code>ERROR in ./src/app/shared/my.service.ts 20:12-21
"export 'AppConfig' was not found in '../app.config'
</code></pre>

App is not served. Havent tried build.

## 🌍 Your Environment
<pre><code>Angular CLI: 8.0.4
Node: 12.4.0
OS: win32 x64
Angular: 8.0.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.4
@angular-devkit/build-angular     0.800.4
@angular-devkit/build-optimizer   0.800.4
@angular-devkit/build-webpack     0.800.4
@angular-devkit/core              8.0.4
@angular-devkit/schematics        8.0.4
@angular/cli                      8.0.4
@angular/pwa                      0.800.4
@ngtools/webpack                  8.0.4
@schematics/angular               8.0.4
@schematics/update                0.800.4
rxjs                              6.5.2
typescript                        3.4.5
webpack                           4.30.0
</code></pre>

**Anything else relevant?**
Had this exact issue almost 3 years ago - https://github.com/angular/angular-cli/issues/2034#issuecomment-246009829. That issue has so many comments that I forget exactly how it was actually resolved, but it was, and warning eventually went away. Now its back.
At that time it was a warning and was ignorable. This time its an error and app is not served...

### PR DESCRIPTION
**Pull request by clydin**
Fixes #14876

### CODE REVIEW

File: packages/ngtools/webpack/src/transformers/ctor-parameters_spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const input = tags.stripIndent`
+      export interface InterInject {};
+      export const INTERFACE_INJECT = new InjectionToken<InterInject>('interface-inject');
+
+      export class MyService {
+        constructor(@Inject(INTERFACE_INJECT) v: InterInject) {}
+      }
+    `;
+
+    const output = `import * as tslib_1 from "tslib"; ; export const INTERFACE_INJECT = new InjectionToken('interface-inject'); let MyService = class MyService { constructor(v) { } }; MyService.ctorParameters = () => [ { type: undefined, decorators: [{ type: Inject, args: [INTERFACE_INJECT,] }] } ]; MyService = tslib_1.__decorate([ tslib_1.__param(0, Inject(INTERFACE_INJECT)) ], MyService); export { MyService };`;
```
Comment (alan-agius4) [inline packages/ngtools/webpack/src/transformers/ctor-parameters_spec.ts]: Nit: can you pls format to make this a bit readable?

Comment (alan-agius4) [inline packages/ngtools/webpack/src/transformers/ctor-parameters_spec.ts]: While its not a problem, it seems at the output contains 2 semi colons.  After the input of tsib

**Comment by alan-agius4** [comment]
test-large CI failures looks legit

File: packages/ngtools/webpack/src/transformers/ctor-parameters.ts [hunk 2]
```diff
@@ -175,6 +177,15 @@ function typeReferenceToExpression(
       return ts.createIdentifier('Number');
     case ts.SyntaxKind.TypeReference:
       const typeRef = node as ts.TypeReferenceNode;
+      const typeSymbol = typeChecker.getSymbolAtLocation(typeRef.typeName);
```
Comment (alan-agius4) [inline packages/ngtools/webpack/src/transformers/ctor-parameters.ts]: Should we care for aliased symbols?

Comment (clydin) [reply packages/ngtools/webpack/src/transformers/ctor-parameters.ts]: Yes.  That's the failure cause.  Although our unit test setup for the transformers doesn't support the necessary file setup to fully test this.  The E2E test provides coverage though.

File: packages/ngtools/webpack/src/transformers/ctor-parameters.ts [hunk 3]
```diff
@@ -175,6 +177,20 @@ function typeReferenceToExpression(
       return ts.createIdentifier('Number');
     case ts.SyntaxKind.TypeReference:
       const typeRef = node as ts.TypeReferenceNode;
+      debugger;
```
Comment (alan-agius4) [inline packages/ngtools/webpack/src/transformers/ctor-parameters.ts]: oops!!!

Comment (clydin) [reply packages/ngtools/webpack/src/transformers/ctor-parameters.ts]: We should really have a lint check for that.

Comment (alan-agius4) [reply packages/ngtools/webpack/src/transformers/ctor-parameters.ts]: Yeah, same for console.log

**Pull request merged**